py_neuromodulation/features/coherence.py

```diff
@@ -83,10 +83,14 @@ def get_coh(self, feature_results, x, y):
         self.Pyy = welch(y, self.sfreq, self.window, nperseg=128)[1]
         self.Pxy = csd(x, y, self.sfreq, self.window, nperseg=128)[1]
 
-        if self.coh:
-            self.coh_val = np.abs(self.Pxy**2) / (self.Pxx * self.Pyy)
-        if self.icoh:
-            self.icoh_val = np.array(self.Pxy / (self.Pxx * self.Pyy)).imag
```

Comment:
Denominator term should have the square root taken. Unfortunately the strategy for coh where you square the numerator and take the absolute value doesn't work for icoh since we extract only the imaginary part and this info isn't properly preserved.